camel/functions/openai_function.py

```diff
@@ -169,72 +169,6 @@ def __init__(
                                    or get_openai_tool_schema(func))
         self.properties = self.openai_tool_schema
 
-    @staticmethod
-    def validate_openai_tool_schema(
-            openai_tool_schema: Dict[str, Any]) -> None:
-        r"""Validates the OpenAI tool schema against
-        :obj:`ToolAssistantToolsFunction`.
-
-        This function checks if the provided :obj:`openai_tool_schema` adheres
-        to the specifications required by OpenAI's
-        :obj:`ToolAssistantToolsFunction`. It ensures that the function
-        description and parameters are correctly formatted according to JSON
-        Schema specifications.
-
-        Args:
-            openai_tool_schema (Dict[str, Any]): The OpenAI tool schema to
-                validate.
-
-        Raises:
-            ValidationError: If the schema does not comply with the
-                specifications.
-            ValueError: If the function description or parameter descriptions
-                are missing in the schema.
-            SchemaError: If the parameters do not meet JSON Schema reference
-                specifications.
-        """
-        # Automatically validates whether the openai_tool_schema passed
-        # complies with the specifications of the ToolAssistantToolsFunction.
-        from openai.types.beta.threads.run import ToolAssistantToolsFunction
-
-        # NOTE: Pydantic v1 does not have a strict mode to check input types
-        # Below is a compromise solution.
-        # refs: https://docs.pydantic.dev/1.10/blog/pydantic-v2/#strict-mode
-        if PYDANTIC_V2:
-            try:
-                ToolAssistantToolsFunction.model_validate(  # type: ignore
-                    openai_tool_schema)
-            except ValidationError as e:
-                raise e
-        elif ToolAssistantToolsFunction(
-                **openai_tool_schema) != openai_tool_schema:
-            raise ValidationError(  # type: ignore
-                errors="Validation error for `ToolAssistantToolsFunction`."
-                "Please check the input data types.",
-                model=ToolAssistantToolsFunction,
-            )
-        # Check the function description
-        if not openai_tool_schema["function"]["description"]:
```

Comment:
`description` and `parameters` validations are still important. Can we only remove line 198 to 215? BTW, I don't think valiadating `openai_tool_schema`, which is generated inside CAMEL, is meaningful.

Reply:
have added `description` and `parameters` validations back